js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.test.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import MinimumOrderCard from './minimum-order-card.js';
+
+jest.mock( '.~/hooks/useAppSelectDispatch' );
+jest.mock( '.~/hooks/useCountryKeyNameMap' );
+jest.mock( '.~/hooks/useStoreCurrency' );
+
+describe( 'MinimumOrderCard', () => {
+	describe( 'onChange callback property', () => {
+		let value, onChange, rendered;
+		beforeEach( () => {
+			value = Object.freeze( [
+				{
+					id: '1',
+					country: 'US',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 20,
+					options: {},
+				},
+				{
+					id: '2',
+					country: 'ES',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 20,
+					options: {
+						free_shipping_threshold: 50,
+					},
+				},
+				{
+					id: '3',
+					country: 'CN',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 25,
+					options: {
+						free_shipping_threshold: 50,
+					},
+				},
+			] );
+			onChange = jest.fn().mockName( 'onChange callback' );
+
+			rendered = render(
+				<MinimumOrderCard value={ value } onChange={ onChange } />
+			);
+		} );
+
+		afterEach( () => {
+			jest.clearAllMocks();
+		} );
+
+		test( 'When the new minimum order value is provided for a remaining country, calls the `onChange` callback with the new value containing `shippingRate.options.free_shipping_threshold` set to the given value', async () => {
+			// Open "Add another…" modal.
+			userEvent.click( rendered.getByRole( 'button', { name: /Add/ } ) );
+			// Input some value.
+			const input = screen.getByRole( 'textbox' );
+			// For some reason it's typed backwards.
+			await userEvent.type( input, '03' );
+			// Confirm.
+			await userEvent.click(
+				screen.getByRole( 'button', {
+					name: /Add minimum order/,
+				} )
+			);
+
+			// Wait for Form component's onSubmit microtask.
+			await waitFor( () => {
+				expect( onChange ).toHaveBeenCalledTimes( 1 );
+				expect( onChange ).toHaveBeenCalledWith( [
+					{
+						...value[ 0 ],
+						options: {
+							free_shipping_threshold: 30,
+						},
+					},
+					value[ 1 ],
+					value[ 2 ],
+				] );
```

Comment:
💅 About the expected value, we are coding the `value[1]` and `value[2]` there. My concern is that if someone were to change the `value` in the `beforeEach` block, say adding another element into the `value` array (so there would be `value[3]`), this test here would then fail, and one needs to add the `value[3]` here to make it pass. 

I would prefer to have something like this, so that it works with the above scenario: 

```js
// this is the expected new value.
const newValue = [ ...value ];
newValue[ 0 ] = {
	...newValue[ 0 ],
	options: {
		free_shipping_threshold: 30,
	},
};

// Wait for Form component's onSubmit microtask.
await waitFor( () => {
	expect( onChange ).toHaveBeenCalledTimes( 1 );
	expect( onChange ).toHaveBeenCalledWith( newValue );
} );
```

Note that this comment also applies to other tests in this file.

Reply:
In https://github.com/woocommerce/google-listings-and-ads/pull/1499/commits/e47c9faa68726d55fb06824588428bb66cffbfba I went with `structuredClone`, to be even less fragile and to emphasize what actually needs to be changed/checked:

```js
// Expect US threshold to be set to 30.
const expectedValue = structuredClone( value );
expectedValue[ 0 ].options.free_shipping_threshold = 30;
expect( onChange ).toHaveBeenCalledWith( expectedValue );
```

Unfortunately, we run an old Node version, so I added a very naive shim, to make it work until we update.

Reply:
> I added a very naive shim

I looked into the implementation (https://github.com/woocommerce/google-listings-and-ads/pull/1499/files#diff-c64e291bfa29797411cbe013a64cc82a4a294cb297c5d546ebb5ca69ff53ffa9R1-R10) and at first I was like 🤯 , and then I realized your mention of "naive" and I'm like okay 😂 Thanks for the code comment there. 

And TIL about [`structuredClone`](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone). Thank you. 🙂 👍

Reply:
I decided to keep it super simple. So instead of putting effort to implement, review, tests, maintaining fully-featured polyfill, we can update Node sooner ;)